src/plugin/file/shp/shpdescriptor.js

```diff
@@ -116,6 +119,30 @@ plugin.file.shp.SHPDescriptor.prototype.clearData = function() {
 };
 
 
+/**
+ * @inheritDoc
+ */
+plugin.file.shp.SHPDescriptor.prototype.getExporter = function() {
+  return new plugin.file.shp.SHPExporter();
+};
+
+
+/**
+ * @inheritDoc
+ */
+plugin.file.shp.SHPDescriptor.prototype.onFileChange = function(options) {
+  plugin.file.shp.SHPDescriptor.base(this, 'onFileChange', options);
+
+  // ensure that the URL is set correctly in case we are converting from a shp/dbf file to a zip shp file
+  const url2 = this.getUrl2();
+  if (url2 && os.file.isLocal(url2)) {
+    const fs = os.file.FileStorage.getInstance();
+    fs.deleteFile(url2);
```

Comment:
Ah good call. Does the SHP already get cleaned up from storage?

Reply:
That's a good point, I'll check. In the other cases when this code is hit, the underlying filename doesn't change, so its just overwritten and uses the same local URL.

Reply:
The .shp file does get overwritten and the .dbf gets removed. Verified by watching IDB from dev tools.